test/ProcessTests.h

```diff
@@ -47,34 +47,92 @@
 
 #include <fstream>
 
-using Mezzanine::String;
-using Mezzanine::Testing::RunCommand;
-using Mezzanine::Testing::GetFileContents;
-
 /// @brief Tests for the class to store test data results.
-AUTOMATIC_TEST_GROUP(ProcessTests, Process)
+BENCHMARK_TEST_GROUP(ProcessTests, Process)
 {
-    // make file and read it back.
-    const Mezzanine::String TestFilename("ProcessTestFile.txt");
-    const Mezzanine::String TestToken(
-                "I've seen things you people wouldn't believe. Attack ships on fire off the "
-                "shoulder of Orion. I watched C-beams glitter in the dark near the Tannhäuser "
-                "Gate. All those moments will be lost in time, like tears in rain. Time to die.");
-    std::ofstream TestFile(TestFilename);
-    TestFile << TestToken;
-    TestFile.close();
-
-    TEST_EQUAL("GetFileContents", TestToken, GetFileContents(TestFilename))
-
-
-    // Try launching a process and reading its stdout
-    TEST_STRING_CONTAINS("RunCommand-stdout",
-                         Mezzanine::String("foo"),
-                         RunCommand("cmake -E echo \"foo\"", "RunCommandScratch.txt"))
-    TEST_THROW("RunCommand-BadCommand",
-               std::runtime_error,
-               []{ RunCommand("echo foo > somefile.txt", "ShouldExistRunCommandScratch.txt"); })
-
+    using namespace Mezzanine;
+    {//GetFileContents
+        const String TestFilename("ProcessTestFile.txt");
+        const String TestToken(
+                    "I've seen things you people wouldn't believe. Attack ships on fire off the "
+                    "shoulder of Orion. I watched C-beams glitter in the dark near the Tannhäuser "
+                    "Gate. All those moments will be lost in time, like tears in rain. Time to die.");
+        std::ofstream TestFile(TestFilename);
+        TestFile << TestToken;
+        TestFile.close();
+
+        TEST_EQUAL("GetFileContents(const_StringView)", TestToken, Testing::GetFileContents(TestFilename))
+    }//GetFileContents
+
+    {//GetCommandOutput
+        TEST_EQUAL("GetCommandOutput(const_StringView)-foo",
+                   String("foo"),
+                   Testing::GetCommandOutput("cmake -E echo foo"))
```

Comment:
You make the testing part of this look so easy.

Reply:
Damn straight.